diff --git a/src/lxplib.c b/src/lxplib.c
index 7200e34..261556c 100644
--- a/src/lxplib.c
+++ b/src/lxplib.c
@@ -517,7 +517,7 @@ static int lxp_stop (lua_State *L) {
   return 1;
 }
 
-static const struct luaL_reg lxp_meths[] = {
+static const struct luaL_Reg lxp_meths[] = {
   {"parse", lxp_parse},
   {"close", lxp_close},
   {"__gc", parser_gc},
@@ -530,7 +530,7 @@ static const struct luaL_reg lxp_meths[] = {
   {NULL, NULL}
 };
 
-static const struct luaL_reg lxp_funcs[] = {
+static const struct luaL_Reg lxp_funcs[] = {
   {"new", lxp_make_parser},
   {NULL, NULL}
 };
